src/daemon/Display.cpp

```diff
@@ -309,32 +309,26 @@ namespace SDDM {
         qDebug() << "Session" << m_sessionName << "selected, command:" << session.exec();
 
         // create new VT for Wayland sessions otherwise use greeter vt
-        int vt = terminalId();
-        if (seat()->name() == QStringLiteral("seat0")) {
-            if (session.xdgSessionType() == QLatin1String("wayland")) {
+        if (seat()->name() == QLatin1String("seat0")) {
+            int vt = terminalId();
+            if (session.xdgSessionType() == QLatin1String("wayland"))
                 vt = VirtualTerminal::setUpNewVt();
-                VirtualTerminal::jumpToVt(vt);
-            }
-        }
+            m_lastSession.setVt(vt);
+            env.insert(QStringLiteral("XDG_VTNR"), QString::number(vt));
```

Comment:
How does this work? 

env isn't created till line 320

Reply:
Just fixed it!